go/internal/feast/onlinestore/cassandraonlinestore.go

```diff
@@ -0,0 +1,404 @@
+	} else {
+		cassandraConfig.password, ok = rawPassword.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert password to string: %v", rawPassword)
+		}
+	}
+
+	// parse keyspace
+	rawKeyspace, ok := onlineStoreConfig["keyspace"]
+	if !ok {
+		cassandraConfig.keyspace = "feast_keyspace"
+		log.Warn().Msg("keyspace not defined: Using 'feast_keyspace' as keyspace instead")
+	} else {
+		cassandraConfig.keyspace, ok = rawKeyspace.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert keyspace to string: %v", rawKeyspace)
+		}
+	}
+
+	// parse protocolVersion
+	protocolVersion, ok := onlineStoreConfig["protocol_version"]
+	if !ok {
+		protocolVersion = 4.0
+		log.Warn().Msg("protocol_version not specified: Using 4 instead")
+	}
+	cassandraConfig.protocolVersion = int(protocolVersion.(float64))
+
+	// parse loadBalancing
+	loadBalancingDict, ok := onlineStoreConfig["load_balancing"]
+	if !ok {
+		loadBalancingDict = gocql.RoundRobinHostPolicy()
+		log.Warn().Msg("no load balancing policy selected, defaulted to RoundRobinHostPolicy")
+	} else {
+		loadBalancingProps := loadBalancingDict.(map[string]any)
+		policy := loadBalancingProps["load_balancing_policy"].(string)
+		switch policy {
+		case "TokenAwarePolicy(DCAwareRoundRobinPolicy)":
+			rawLocalDC, ok := loadBalancingProps["local_dc"]
+			if !ok {
+				return nil, fmt.Errorf("a local_dc is needed for policy DCAwareRoundRobinPolicy")
+			}
+			localDc := rawLocalDC.(string)
+			cassandraConfig.loadBalancingPolicy = gocql.TokenAwareHostPolicy(gocql.DCAwareRoundRobinPolicy(localDc))
+		case "DCAwareRoundRobinPolicy":
+			rawLocalDC, ok := loadBalancingProps["local_dc"]
+			if !ok {
+				return nil, fmt.Errorf("a local_dc is needed for policy DCAwareRoundRobinPolicy")
+			}
+			localDc := rawLocalDC.(string)
+			cassandraConfig.loadBalancingPolicy = gocql.DCAwareRoundRobinPolicy(localDc)
+		default:
+			log.Warn().Msg("defaulted to using RoundRobinHostPolicy")
+			cassandraConfig.loadBalancingPolicy = gocql.RoundRobinHostPolicy()
+		}
+	}
+
+	// parse connectionTimeoutMillis
+	connectionTimeoutMillis, ok := onlineStoreConfig["connection_timeout_millis"]
+	if !ok {
+		connectionTimeoutMillis = 0.0
+		log.Warn().Msg("connection_timeout_millis not specified, using gocql default")
+	}
+	cassandraConfig.connectionTimeoutMillis = int64(connectionTimeoutMillis.(float64))
+
+	// parse requestTimeoutMillis
+	requestTimeoutMillis, ok := onlineStoreConfig["request_timeout_millis"]
+	if !ok {
+		requestTimeoutMillis = 0.0
+		log.Warn().Msg("request_timeout_millis not specified, using gocql default")
+	}
+	cassandraConfig.requestTimeoutMillis = int64(requestTimeoutMillis.(float64))
+
+	// parse numConnections
+	numConnections, ok := onlineStoreConfig["num_connections"]
+	if !ok {
+		numConnections = 0.0
+		log.Warn().Msg("num_connections not specified, using gocql default")
+	}
+	cassandraConfig.numConnections = int(numConnections.(float64))
```

Comment:
if ok, set this and log INFO message with value. We can get rid of the !ok case and check for nil when using the value.